assets/javascripts/needleeditor.js

```diff
@@ -545,12 +500,49 @@ function saveNeedle(overwrite) {
   document.getElementById('save').disabled = true;
   document.getElementById('needleeditor_overwrite').value = overwrite ? '1' : '0';
 
-  $.ajax({
-    type: 'POST',
-    url: form.attr('action'),
-    data: form.serialize(),
-    complete: reactToSaveNeedle
-  });
+  fetchWithCSRF(form.action, {method: 'POST', body: new FormData(form)})
+    .then(response => {
+      if (!response.ok) throw `Server returned ${response.status}: ${response.statusText}`;
+      return response.json();
+    })
+    .then(response => {
+      if (response.error) throw response.error;
+      if (response.requires_overwrite) {
+        delete response.requires_overwrite;
+        const modalElement = document.getElementById('modal-overwrite');
+        modalElement.dataset.formdata = response;
+        modalElement.getElementsByClassName('modal-title')[0].textContent = `Sure to overwrite ${response.needlename}?`;
+        if (!window.overwriteModal) {
+          window.overwriteModal = new bootstrap.Modal(modalElement);
+        }
+        window.overwriteModal.show();
+      } else if (response.success) {
+        // add note to go back or restart
+        if (response.developer_session_job_id) {
+          response.success +=
+            " - <a href='" +
+            urlWithBase('/tests/' + response.developer_session_job_id) +
+            "#live'>back to live view</a>";
+        } else if (response.restart) {
+          response.success += " - <a href='#' data-url='" + response.restart + "' class='restart-link'>restart job</a>";
+        }
+        addFlash('info', response.success);
+      } else {
+        throw `<b>Unknown Error:</b><code>$response</code>`;
```

Comment:
In JavaScript you have to use curly brackets:
```suggestion
        throw `<b>Unknown Error:</b><code>${response}</code>`;
```